CMakeLists.txt

```diff
@@ -615,4 +627,29 @@ else ()
   endif ()
 endif (IS_64BIT_BUILD EQUAL 1)
 
+
+# The prebuilt IDL headers aren't suitable for 32 bit builds, attempt to use the ones 
+# generated when building corguids by including them before the prebuilt ones
+if (WIN32 AND CLR_CMAKE_PLATFORM_ARCH_I386)
```

Comment:
If I use the prebuilt headers then I get the following error while building cee_wks:

> `d:\projects\coreclr\src\vm\cominterfacemarshaler.cpp(169): error C2664: 'HRESULT IManagedObject::GetObjectIdentity(BSTR *,int *,__int64 *)' : cannot convert argument 3 from 'INT_PTR *' to '__int64 *' [D:\Projects\m
       coreclr\bin\obj\Windows_NT.x86.Debug\src\vm\wks\cee_wks.vcxproj]`

The issue is that in the prebuilt mscoree.h the type of `pCCW` parameter declared by `IManagedObject::GetObjectIdentity` is `__int64`. cominterfacemarshaler.cpp uses `INT_PTR` so it works on x64 but not on x86.

mscoree.idl contains the following:

```C++
#ifdef _WIN64
    #define CCW_PTR __int64 *
#else // WIN64
    #define CCW_PTR int *
#endif // WIN64
```

so you get different headers out of the IDL file depending on the target arch.

To make things even more confusing
* mscoree.h that ships with the Windows SDK uses `int` for this parameter so it appears that it is x86 specific. But I don't remember ever seeing arch specific headers in the SDK before.
* the MSDN [documentation](https://msdn.microsoft.com/en-us/library/aa964958(v=vs.110).aspx) says that the type of the parameter is CCW_PTR but that type didn't made it to the header file.

Something seems wrong but I don't know what and where.